### PR TITLE
Add mechanism to allow building partly underground

### PR DESCRIPTION
**Pull request by Gymnasiast**
Normally, the game requires you to build a whole piece either underground or above ground, but doesn’t allow you to mix.

However, provided the shape of the piece is right, and the relevant tunnel code has been added, this is perfectly possible:

afbeelding

The heartline roll is 5 blocks long and draws perfectly, even when a vehicle goes through it.

This PR introduces the flag and adds it for the heartline rolls. For others pieces, it requires modification of the drawing code, which is something to do on a piece-by-piece basis. This is also why I didn’t yet add a changelog (I wanted to wait until more of them were added), but I can of course change that.

### CODE REVIEW

File: src/openrct2/ride/TrackData.cpp [hunk 1]
```diff
@@ -5956,7 +5956,7 @@ static constexpr uint8_t TrackSequenceElementAllowedWallEdges[][MaxSequencesPerP
 static_assert(std::size(TrackSequenceElementAllowedWallEdges) == TrackElemType::Count);
 
 /** rct2: 0x0099423C */
-static constexpr uint16_t TrackFlags[] = {
```
Comment (tupaschoal) [inline src/openrct2/ride/TrackData.cpp:5959]: Is this change needed? Weere we at the verge of overflow?

Comment (Gymnasiast) [reply src/openrct2/ride/TrackData.cpp:5959]: Yes, `TRACK_ELEM_FLAG_CAN_BE_PARTLY_UNDERGROUND` did not fit into a uint16_t, so I had to enlarge it.

**Review by tupaschoal — COMMENTED** [review]
What happens if it is sloped just at the height the piece will be built?

I couldn't follow the logic to make sure it is not allowing for too much

**Comment by Gymnasiast** [comment]
I don’t quite understand, could you elaborate?

**Review by tupaschoal — COMMENTED** [review]
> I don’t quite understand, could you elaborate?

Nevermind, I downloaded and tested it.

This PR does need a changelog entry and a network bump, right?

**Review by tupaschoal — APPROVED** [review]
